recipes/newrelic/apm/java/linux.yml

```diff
@@ -49,18 +58,31 @@ preInstall:
       apt-get install -y -qq ./nri-introspector-java_0.1.0_SNAPSHOT-2_amd64.deb 2> /dev/null || true
     fi
 
-    nri-lsi-java -list | tr -d '[]' | tr ',' '\n' > $tmpDir/processes
-    for JAVA_PID in $(cat $tmpDir/processes)
+    # check if any tomcat instances are service-managed
+    nri-lsi-java -list | tr -d '[]' | tr ',' '\n' > $TMP_DIR/all_java_processes
+    for JAVA_PID in $(cat $TMP_DIR/all_java_processes)
     do
       introspectionData=$(nri-lsi-java -introspect ${JAVA_PID})
-      mainClass=$( echo "$introspectionData" | grep -oP '.*\K(?<=mainClass":").*?(?=\")' )
-      if [[ "$mainClass" == "org.apache.catalina.startup.Bootstrap" ]]; then
-        rm -rf $tmpDir
-        exit 0
+      MAIN_CLASS=$( echo "$introspectionData" | grep -oP '.*\K(?<=mainClass":").*?(?=\")' )
+      if [[ "$MAIN_CLASS" == "org.apache.catalina.startup.Bootstrap" ]]; then
+        MANAGED_TOMCAT=$(systemctl status $JAVA_PID | head -n1 | grep -oP "(?<=. )\K.+(?=\.service)" | wc -l)
+        if [[ $MANAGED_TOMCAT -eq 0 ]]; then
+          touch $TMP_DIR/tomcat_standalone_process
+        else
+          rm -rf $TMP_DIR
+          exit 0
+        fi
       fi
     done
 
-    rm -rf $tmpDir
+    if [[ -f $TMP_DIR/tomcat_standalone_process ]]; then
+      echo -e "All detected Tomcat processes are running external to systemd" >&2
+    else
+      echo -e "Tomcat processes were detected, but no running java processes found" >$2
+    fi
+
+
+    rm -rf $TMP_DIR
```

Comment:
When we exit here, we know there is no managed tomcat, and the lsi has been installed.
In this case the recipe won't be running (what we want).
I think we should also cleanup the installed lsi and run a yum/apt-get remove nri-lsi-java to leave the host in the same state as we found it before evaluating if the recipe should run.
Here is an example https://github.com/newrelic/open-install-library/blob/feat/java-dynamic/recipes/newrelic/apm/java/linux.yml#L106-L110